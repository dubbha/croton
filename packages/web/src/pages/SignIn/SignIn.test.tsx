import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

describe('pages/SignIn', () => {
  it('should render successfully', () => {
    jest.isolateModules(() => {
      jest.doMock('react-redux', () => ({
        useDispatch: () => jest.fn(),
        useSelector: () => false,
      }));

      const { SignIn } = require('./SignIn');

      const { container } = render(<SignIn />, { wrapper: MemoryRouter });
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should dispatch on submit', () => {
    jest.isolateModules(() => {
      const fn = jest.fn();
      jest.doMock('react-redux', () => ({
        useDispatch: () => fn,
        useSelector: () => false,
      }));

      jest.doMock('components/SignInForm', () => {
        const React = require('react');
        return {
          SignInForm: ({ onSubmit }) => (
            <button onClick={onSubmit} data-testid="submitButton" />
          ),
        };
      });

      const { SignIn } = require('./SignIn');

      const { getByTestId } = render(<SignIn />, { wrapper: MemoryRouter });
      const submitButton = getByTestId('submitButton');

      fireEvent.click(submitButton);
      expect(fn).toBeCalled();
    });
  });
});
