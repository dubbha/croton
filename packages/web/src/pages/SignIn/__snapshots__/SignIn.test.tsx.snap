// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/SignIn should render successfully 1`] = `
<div
  class="container"
>
  <header>
    <nav
      class="navbar navbar-expand-lg navbar-light"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="logo"
            class="d-inline-block align-top"
            height="30"
            src="logo.svg"
            width="30"
          />
           
          <span
            class="companyName"
          >
            Croton
          </span>
        </a>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="#features"
            href="#features"
          >
            Features
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="#about"
            href="#about"
          >
            About
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            data-rb-event-key="#services"
            href="#services"
          >
            Services
          </a>
        </div>
      </div>
      <div
        class="navbar-nav"
      >
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/signin"
          >
            Sign In
          </a>
        </div>
        <div
          class="nav-item"
        >
          <a
            class="nav-link"
            href="/signup"
          >
            Sign Up
          </a>
        </div>
      </div>
    </nav>
  </header>
  <div
    class="container signInConrollersContainer"
  >
    <form
      class="signinForm"
    >
      <div
        style="height: 50px; margin-bottom: 16px;"
      />
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          Email address
        </label>
        <input
          class="form-control"
          data-testid="signInForm__email"
          placeholder="Enter email"
          type="email"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
        >
          Password
        </label>
        <input
          class="form-control"
          data-testid="signInForm__password"
          placeholder="Enter password"
          type="password"
          value=""
        />
        <small
          class="reset-password-link form-text"
        >
          <a
            href="/reset"
          >
            Forgot password?
          </a>
        </small>
      </div>
      <button
        class="btn btn-primary"
        data-testid="submitButton"
        disabled=""
        type="submit"
      >
        <div
          class="spinner-container"
        />
        <span>
          Sign In
        </span>
      </button>
    </form>
    <button />
  </div>
  <div
    class="signin-register"
  >
    Don't have an account? 
    <a
      href="/signup"
    >
       Register
    </a>
  </div>
  <footer>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="copyright"
          >
            Â© 
            2020
             
            <a
              href="https://github.com/dubbha/croton"
            >
              Croton
            </a>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="justify-content-end nav"
          >
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                data-rb-event-key="https://opensource.org/licenses/MIT"
                href="https://opensource.org/licenses/MIT"
                target="_blank"
              >
                MIT License
              </a>
            </div>
            <div
              class="nav-item"
            >
              <a
                class="nav-link"
                href="#"
                role="button"
                target="_blank"
              >
                Contact Us
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
